### DISCUSSION TITLE
12. How do we ensure that the solutions being developed today, which we build dependencies on, will maintain long term viability, sustainability, maintainability?

### DISCUSSION THREAD

**Discussion by MichaelRimler** [phuse-org/OSTCDA#12] [General]
What if people retire, move on to other companies, leave the industry - how do we manage the risk?

**Comment by KatjaGlassConsulting** [comment]
- active commitment and funding (e.g. time resources)
- quality measurements (is the code understandable & maintainable by others)
- collaborations on successors if required

**Comment by MichaelRimler** [comment]
This is a big question that I don't think is being addressed sufficiently at the moment. Industry is gravitating towards R for clinical data analysis. There is investment in new R-based solutions such as [admiral](https://github.com/pharmaverse/admiral), [pkglite](https://github.com/Merck/pkglite), [teal](https://github.com/insightsengineering/teal), [tfrmt](https://github.com/GSK-Biostatistics/tfrmt) - but what if a package owner / maintainer stops supporting. Or the organization(s) maintaining make a decision to orphan the package. One of the beauties of permissively licensed, open-source code software is the ability for anyone to adopt the orphaned code and begin to maintain again. But, there is a risk that organizations embed an open source solution into their clinical data pipeline processes and it 'disappears' because of breaking changes of dependencies, low or no maintenance of the package team, or other reasons. How will this be managed?

**Comment by orladoylenvs** [comment]
This is a great point. Maybe, more broadly stated, how can we ensure that life science companies are not purely open source users with little contribution to the maintenance of packages. 
Closer cross-industry collaboration will be essential to establish solid foundations for packages, many of which are emerging. 
Some ideas on the ground could be to have dedicated sprint/s in your mission where the team focuses on addressing key issues or in a KanBan setting maybe an issue buffer? Making these contributions visible cross-industry may help ensure we follow through?

**Comment by rossfarrugia** [comment]
I'm hoping via communities and groups like pharmaverse, PHUSE, R Consortium we could find new owners to take over maintenance and ownership of key OS code if we ever needed to. The key is getting the code well embedded across the industry and enough willingness across companies to continue to contribute, as if it ever ends up only 1 or 2 companies carrying all the weight then it won't be sustainable. Once more companies feel the benefits of using OS, you'd hope they give back by making space for their individuals to contribute.

**Comment by chowsanthony** [comment]
The issue of underfunding for Load-Bearing Internet People (LBIPs) remains serious all these years. LBIPs are those people maintaining essential internet infrastructure. Many LBIPs work as unpaid volunteers, risking burnout and personal hardship. Traditional funding models fail to support them adequately.
